### PR TITLE
Network type serialization rearrangement

### PR DESCRIPTION
**Pull request by STMiki**
Pls ?

### CODE REVIEW

**Comment by huntears** [comment]
I see that you used `#pragma once`, but we have the rest of the codebase using `#indef` macros.
I would suggest changing it to that.

File: cubic-server/Chat.cpp [hunk 1]
```diff
@@ -240,6 +235,87 @@ nlohmann::json chat::message::HoverEvent::toJson() const
...
+        throw std::runtime_error("Unknown click event action: " + action);
+    event.value = json["value"].get<std::string>();
+    return event;
+}
+
+chat::message::HoverEvent chat::message::HoverEvent::fromJson(const nlohmann::json &json)
+{
+    HoverEvent event;
+    std::string action = json["action"].get<std::string>();
+    if (action == "show_text")
```
[the remark this answers is not in the document]
Comment (STMiki) [reply cubic-server/Chat.cpp:307]: I really REALLY would like to comply, but switch case on std::string is just too much of a trouble T.T

File: cubic-server/protocol/typeSerialization.hpp [hunk 2]
```diff
@@ -1,341 +1,15 @@
...
+#include "serialization/addPrimaryType.hpp"
+#include "serialization/add.hpp"
+#include "serialization/popPrimaryType.hpp"
+#include "serialization/pop.hpp"
+
+// bitset = std::vector<uint64_t>
+// ! could throw if if we access a bit that is not in the bitset
+#define BITSET_GET_BIT(bitset, bit) (bitset[bit / 64] & (1ULL << (bit % 64)) == 1)
+// ! could throw if if we access a bit that is not in the bitset
+#define BITSET_SET_BIT(bitset, bit, value) bitset[bit / 64] = (bitset[bit / 64] & ~(1ULL << (bit % 64))) | (value << (bit % 64))
```
Comment (huntears) [inline cubic-server/protocol/typeSerialization.hpp:13]: Use constexpr for this

**Review by huntears — CHANGES_REQUESTED** [review]
Serialization folder not included in CMakeLists.txt

**Review by huntears — APPROVED** [review]

**Pull request merged**